package-kuring/Sources/Features/BotFeatures/BotFeature.swift

```diff
@@ -0,0 +1,158 @@
+    }
+    
+    public enum Action: BindableAction, Equatable {
+        case binding(BindingAction<State>)
+        case sendMessage
+        case messageResponse(Result<String, ChatError>)
+        case addQuestion(String)
+        case queryChanged([ChatInfo])
+        case onAppear
+        
+        public enum ChatError: Error, Equatable {
+            case serverError(Error)
+            
+            public static func == (lhs: ChatError, rhs: ChatError) -> Bool {
+                switch (lhs, rhs) {
+                case let (.serverError(lhsError), .serverError(rhsError)):
+                    return lhsError.localizedDescription == rhsError.localizedDescription
+                }
+            }
+        }
+    }
+    
+    @Dependency(\.bots) public var context
+    
+    public var body: some ReducerOf<Self> {
+        BindingReducer()
+        Reduce { state, action in
+            switch action {
+            case .binding:
+                return .none
+                
+            case .sendMessage:
+                state.focus = nil
+                state.isLoading = true
+                return .run { [question = state.chatInfo.text] send in
+                    do {
+                        SSEClient.shared.sessionStart(question: question)
+                        
+                        /// 작업이 취소되거나 완료되기 전까지 send가 호출되지 않도록 보장
+                        await withTaskCancellationHandler {
+                            let continuation = AsyncStream<String>.Continuation.self
+                            
+                            ///서버로부터 받은 메시지들을 비동기 스트림으로 처리
+                            let stream = AsyncStream<String> { continuation in
+                                SSEClient.shared.sendMessage = { message in
+                                    continuation.yield(message)
+                                }
+                            }
+                            
+                            for await message in stream {
+                                await send(.messageResponse(.success(message)))
+                            }
+                            
+                        } onCancel: {
+                            SSEClient.shared.task?.cancel()
+                        }
+                        
+                    } catch {
+                        await send(.messageResponse(.failure(.serverError(error))))
+                    }
+                }
+                
+            case let .messageResponse(.success(message)):
+                state.isLoading = false
+                if let lastMessage = state.chatHistory.last, lastMessage.type == .answer {
+                    state.chatHistory[state.chatHistory.count - 1].text += message
+                    state.chatHistory[state.chatHistory.count - 1].chatStatus = .complete
+                }
+                return .none
+                
+            case let .messageResponse(.failure(error)):
+                print(error.localizedDescription)
```

Comment:
여기는 로딩 상태 해지 안해도 되나요?
```suggestion
                state.isLoading = false
                print(error.localizedDescription)
```

Reply:
앗 수정하겠습니다 !